auth/auth-server/src/telemetry/quote_comparison/mod.rs

```diff
@@ -10,9 +11,12 @@ const DECIMAL_TO_BPS: f64 = 10_000.0;
 
 /// Represents a single quote comparison between quotes from different sources
 pub struct QuoteComparison<'a> {
```

Comment:
First time I'm looking at this & it's out of scope for this PR / review, but why do we hold references (`&'a QuoteResponse`) instead of owned types (`QuoteResponse`) here? Generally find this to make things much less ergonomic due to managing lifetimes.

`QuoteResponse` is a pretty lightweight struct, it should be totally fine to clone it into `QuoteComparison` if ownership can't be taken directly